### PR TITLE
Add circleci testing of frontend/backend + deployment

### LINKED ISSUE #68: Add circleci automatic deployment / Dockerfile building
- [ ] Figure out an automated api deployment strategy with circleci as the production machine we        are using does not support external sshing (potentially moving to digitialocean).
- [ ] Utilize circleci to build the `acmweb/api` Dockerfile instead of Docker automated builds for speed and convenience.

### PR DESCRIPTION
**Pull request by kevinschoonover**
# Pull Request Description
Closes #68 

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -2,19 +2,54 @@
...
-version: 2
-jobs:
-  build:
+version: 2.1
+executors:
+  docker-publisher:
+    environment:
+      IMAGE_NAME: acmweb/api
+    docker:
+      - image: circleci/buildpack-deps:stable
```
Comment (ClayMav) [inline .circleci/config.yml]: lets version lock this as well

**Comment by kevinschoonover** [comment]
@ClayMav ready

File: apps/api/README.md [hunk 2]
```diff
@@ -38,42 +38,68 @@ To get a local copy up and running follow these simple steps.
...
+        cp ./.docker/web.env.default ./.docker/web.env
+        # Now edit the ./.docker/web.env with the appropriate values.
+        # Instructions how are located in the file.
+        ```
+
 5. Start the docker containers using `docker-compose`:
-```sh
-docker-compose up
-```
+    ```sh
```
Comment (ClayMav) [inline apps/api/README.md:90]: note that you must be in apps

File: deploy/README.md [hunk 3]
```diff
@@ -33,133 +24,179 @@ To get a local copy up and running follow these simple steps.
...
-    appear, this will be important for ansible):
-    ```sh
-    terraform apply
+2. Switch to the `production` terraform workspace:
+    ```
+    terraform workspace select production
     ```
 
-15. Copy `group_vars/api-prod.yml.template` into `group_vars/api-prod.yml`:
+2. Run terraform:
```
Comment (ClayMav) [inline deploy/README.md:153]:
```suggestion
3. Run terraform:
```

File: deploy/README.md [hunk 4]
```diff
@@ -33,133 +24,179 @@ To get a local copy up and running follow these simple steps.
...
-    ![Database Page](./imgs/digitalocean_database_page.png)
-
-18. Configure the **Trusted Sources** to be the droplet created with terraform
-    `api.mstacm.org`. If completed correctly, this should look like:
-    ![Trusted Sources](./imgs/trusted_sources.png)
-
-19. Create the `phoenix` database and user in the **Users & Databases** tab.
-
-20. (optional, but recommended) Upload an old backup of the database to the new
+15. (optional, but recommended) Upload an old backup of the database to the new
```
Comment (ClayMav) [inline deploy/README.md:158]:
```suggestion
4. (optional, but recommended) Upload an old backup of the database to the new
```

File: deploy/README.md [hunk 5]
```diff
@@ -33,133 +24,179 @@ To get a local copy up and running follow these simple steps.
...
     cat <YOUR_BACKUP_FILE> | gzip -d | psql <YOUR_CONNETION_STRING_TO_DATABASE>
     ```
 
-21. Edit the `groups_vars/api-prod.yml` file with the appropriate environment
-    variables. Tips on where the acquire the correct values are found in the
-    comments of this file.
+    The connection string can be found in the online database panel on
+    digitalocean.
+
+16. Edit the `groups_vars/secrets.yml` file with the appropriate environment
```
Comment (ClayMav) [inline deploy/README.md:167]:
```suggestion
5. Edit the `groups_vars/secrets.yml` file with the appropriate environment
```

File: deploy/README.md [hunk 6]
```diff
@@ -33,133 +24,179 @@ To get a local copy up and running follow these simple steps.
...
+    using [Ansible Vault][ansible-vault-url] so that it can be automatically
+    deployed by [Circle CI][circle-ci-url]. To edit, get the vault password from
+    the ACM Bitwarden account or ask the Chair of ACM Web and use the following
+    command:
+    ```
+    poetry run ansible-vault edit group_vars/secrets.yml
+    ```
     
-22. Run the ansible playbook:
+17. Run the ansible playbook:
```
Comment (ClayMav) [inline deploy/README.md:180]:
```suggestion
6. Run the ansible playbook:
```

**Pull request merged**